src/Workspaces/Core/Portable/SymbolId/SymbolKey.TupleTypeSymbolKey.cs

```diff
@@ -34,35 +34,32 @@ public override SymbolKeyResolution Resolve(Compilation compilation, bool ignore
                 Compilation compilation,
                 bool ignoreAssemblyKey)
             {
-                // We need all types to have a resolution and we ignore ambiguous candidates
-                ITypeSymbol[] types = _types.Select(a => a.Resolve(compilation, ignoreAssemblyKey).Symbol as ITypeSymbol).ToArray();
-                if (types.Any(a => a == null))
+                INamedTypeSymbol underlyingType = _underlyingType.Resolve(compilation, ignoreAssemblyKey).Symbol as INamedTypeSymbol;
+                if ((object)underlyingType == null)
                 {
                     return SpecializedCollections.EmptyEnumerable<INamedTypeSymbol>();
                 }
 
                 if (_names == null)
                 {
-                    return SpecializedCollections.SingletonEnumerable(compilation.CreateTupleTypeSymbol(types.ToImmutableArray()));
+                    return SpecializedCollections.SingletonEnumerable(compilation.CreateTupleTypeSymbol(underlyingType));
                 }
                 else
                 {
-                    return SpecializedCollections.SingletonEnumerable(compilation.CreateTupleTypeSymbol(types.ToImmutableArray(), _names.ToImmutableArray()));
+                    return SpecializedCollections.SingletonEnumerable(compilation.CreateTupleTypeSymbol(underlyingType, _names.ToImmutableArray()));
                 }
             }
 
             internal override bool Equals(TupleTypeSymbolKey other, ComparisonOptions options)
             {
-                var comparer = SymbolKeyComparer.GetComparer(options);
-
-                return SequenceEquals(other._types, _types, comparer)
+                return _underlyingType.Equals(other._underlyingType, options) 
                        && SequenceEquals(other._names, _names, StringComparer.Ordinal);
             }
 
             internal override int GetHashCode(ComparisonOptions options)
             {
-                // Types are good enough for hash code, we don't need to include names.
-                return Hash.CombineValues(_types);
+                // The hash of the underlying type is good enough, we don't need to include names.
```

Comment:
This seems odd.  It does mean that effectively all tuples of the same length will hash to the same entity right?  That concerns me (but only slightly).